src/Scripting/CSharp/CSharpScript.cs

```diff
@@ -3,45 +3,33 @@
 using System;
 using System.Diagnostics;
 using System.Globalization;
-using System.Text;
-using Microsoft.CodeAnalysis;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.CSharp;
-using Microsoft.CodeAnalysis.Text;
 
 namespace Microsoft.CodeAnalysis.Scripting.CSharp
 {
     /// <summary>
     /// A factory for creating and running csharp scripts.
     /// </summary>
-    public sealed class CSharpScript : Script
+    public static class CSharpScript
     {
-        private CSharpScript(string code, string path, ScriptOptions options, Type globalsType, Type returnType, ScriptBuilder builder, Script previous)
-            : base(code, path, options, globalsType, returnType, builder, previous)
-        {
-        }
-
-        internal override Script Make(string code, string path, ScriptOptions options, Type globalsType, Type returnType, ScriptBuilder builder, Script previous)
-        {
-            return new CSharpScript(code, path, options, globalsType, returnType, builder, previous);
-        }
-
         /// <summary>
         /// Create a new C# script.
         /// <param name="code">The source code of the script.</param>
         /// <param name="options">The script options.</param>
         /// </summary>
-        public static Script Create(string code, ScriptOptions options)
+        public static Script<T> Create<T>(string code, ScriptOptions options)
```

Comment:
You should also have overloads that are not generic and just imply the type as object.